sklearn/metrics/_pairwise_distances_reduction/meson.build

```diff
@@ -34,6 +34,7 @@ _datasets_pair = py.extension_module(
   dependencies: [np_dep, openmp_dep],
   override_options: ['cython_language=cpp'],
   cython_args: cython_args,
+  cpp_args: ['-std=c++11'],
```

Comment:
@jjerphan @Micky774 comments here?

Reply:
I did a bit of archaeology and the std=c++11 comes from https://github.com/scikit-learn/scikit-learn/pull/22320, but I did not find any explanations about why it was needed.

This may be related to sized deallocation according to https://stackoverflow.com/questions/53022608/application-crashes-with-symbol-zdlpvm-version-qt-5-not-defined-in-file-libqt

Reply:
This is required to use elements of the C++ standard library, such as `std::vector` and `std::move`.

Reply:
OK this was probably the original motivation. The fact that there is a missing symbol with `-std=c++14` (see https://github.com/scikit-learn/scikit-learn/pull/28506#discussion_r1512461087) is another unrelated issue ... not sure if this is worth looking into or not.

Reply:
I would feel more comfortable if our modern code didn't require c++11 instead of 14 though. But not related to this PR.

Reply:
As always, I would like to understand the underlying problem, but I am not sure when I will get some the time.

Reply:
The problem seems to happen only in `_argkmin_classmode` for some reason:

Without `std=c++11`, symbol is `_ZdlPv` and is missing:
```
❯ objdump ~/micromamba/envs/del/lib/python3.12/site-packages/sklearn/metrics/_pairwise_distances_reduction/_argkmin_classmode.cpython-312-x86_64-linux-gnu.so -x | grep _ZdlPv
0000000000000000         *UND*	0000000000000000              _ZdlPv
0000000000031660 g     F .text	0000000000000005              _ZdlPvm
```

With `std=c++11`, symbol is `_ZdlPv@@GLIBCXX_3.4`:
```
❯ objdump ~/micromamba/envs/del/lib/python3.12/site-packages/sklearn/metrics/_pairwise_distances_reduction/_argkmin_classmode.cpython-312-x86_64-linux-gnu.so -x | grep _ZdlPv
0000000000000000       F *UND*	0000000000000000              _ZdlPv@@GLIBCXX_3.4
```

Reply:
So this seems to be really about sized deallocation i.e. I am only using `-fno-sized-deallocation` in my last commit with c++14 and it fixes the issue.

No idea what to make of this but this may be useful to someone with more C++ technical skills than me.

Reply:
Yeah this is new territory for me too. I can't wait until the inevitable stack exchange post that explains this in tremendous detail in a few years. Thanks for looking into this 🙏